automower_ble/mower.py

```diff
@@ -11,375 +11,105 @@
-        )
-        await self.client.connect()
-        logger.info("connected")
-
-        logger.info("pairing device...")
-        await self.client.pair()
-        logger.info("paired")
-
-        self.client._backend._mtu_size = MTU_SIZE
-
-        for service in self.client.services:
-            logger.info("[Service] %s", service)
-
-            for char in service.characteristics:
-                if "read" in char.properties:
-                    try:
-                        value = await self.client.read_gatt_char(char.uuid)
-                        logger.debug(
-                            "  [Characteristic] %s (%s), Value: %r",
-                            char,
-                            ",".join(char.properties),
-                            value,
-                        )
-                    except Exception as e:
-                        logger.error(
-                            "  [Characteristic] %s (%s), Error: %s",
-                            char,
-                            ",".join(char.properties),
-                            e,
-                        )
-
-                else:
-                    logger.debug(
-                        "  [Characteristic] %s (%s)", char, ",".join(char.properties)
-                    )
-                if char.uuid == "98bd0002-0b0e-421a-84e5-ddbf75dc6de4":
-                    self.write_char = char
-
-                if char.uuid == "98bd0003-0b0e-421a-84e5-ddbf75dc6de4":
-                    self.read_char = char
-
-        async def notification_handler(
-            characteristic: BleakGATTCharacteristic, data: bytearray
-        ):
-            logger.info("Received: " + str(binascii.hexlify(data)))
-            await self.queue.put(data)
-
-        await self.client.start_notify(self.read_char, notification_handler)
-
-        await asyncio.sleep(5.0)
-
-        request = self.request.generate_request_setup_channel_id()
-        response = await self._request_response(request)
-        if response == None:
-            return False
-
-        ### TODO: Check response
-
-        request = self.request.generate_request_handshake()
-        response = await self._request_response(request)
-        if response == None:
-            return False
-
-        ### TODO: Check response
-
-        if self.pin is not None:
-            request = self.request.generate_request_pin(self.pin)
-            response = await self._request_response(request)
-            if response == None:
-                return False
-
-            ### TODO: Check response
-
-        return True
+        This is the same function as get_parameter but with a differerent name to make syntax a bit more clear.
+        It also does not handle any response even though it upstream reads the response."""
+        self.get_parameter(parameter_name, **kwargs)
 
-    def is_connected(self) -> bool:
-        return self.client.is_connected
-
-    async def probe_gatts(self, device):
```

Comment:
We should keep this function. It's used by the HA integration to read information without pairing

Reply:
It's kept in the parent BLEClient class.